### PR TITLE
feat: Package controller

### PR DESCRIPTION
**Pull request by yashvardhan-kukreja**
Signed-off-by: Yashvardhan Kukreja <ykukreja@redhat.com>

### CODE REVIEW

File: cmd/package-operator-manager/main.go [hunk 1]
```diff
@@ -147,7 +148,7 @@ func run(log logr.Logger, scheme *runtime.Scheme, opts opts) error {
 	dc := dynamiccache.NewCache(
 		mgr.GetConfig(), mgr.GetScheme(), mgr.GetRESTMapper(), recorder,
 		dynamiccache.SelectorsByGVK{
-			// Only cache objects with our label selector,
+			// Only cache objects with our label selector,=
```
Comment (garrettthomaskth) [inline cmd/package-operator-manager/main.go:151]:
```suggestion
			// Only cache objects with our label selector,
```

File: config/images/package-loader.Containerfile [hunk 2]
```diff
@@ -1,9 +1,9 @@
-FROM scratch
+FROM registry.access.redhat.com/ubi8/ubi-minimal@sha256:574f201d7ed185a9932c91cef5d397f5298dff9df08bc2ebb266c6d1e6284cd1
```
Comment (garrettthomaskth) [inline config/images/package-loader.Containerfile:1]: Should this be done for all images in `config/`?

Comment (yashvardhan-kukreja) [reply config/images/package-loader.Containerfile:1]: ohh whoops! I missed to revert this.

A few days back I used the ubi-minimal image for some blocker but that blocker sorted out and I forgot to revert it back to scratch.

Will do that :)

File: internal/controllers/controllers.go [hunk 3]
```diff
@@ -22,16 +22,23 @@ const (
...
-// Ensures the given finalizer is set and persisted on the given object.
-func EnsureFinalizer(
+// Ensures the given finalizers are set and persisted on the given object.
+func EnsureFinalizers(
 	ctx context.Context, c client.Client,
-	obj client.Object, finalizer string,
+	obj client.Object, finalizers ...string,
 ) error {
-	if controllerutil.ContainsFinalizer(obj, finalizer) {
+	noFinalizersToPatch := true
```
Comment (garrettthomaskth) [inline internal/controllers/controllers.go:30]:
```suggestion
	if len(finalizers) == 0 {
		return nil
	}
```

File: internal/controllers/controllers.go [hunk 4]
```diff
@@ -49,17 +56,23 @@ func EnsureFinalizer(
...
-// Removes the given finalizer and persists the change.
-func RemoveFinalizer(
+// Removes the given finalizers and persist the change.
+func RemoveFinalizers(
 	ctx context.Context, c client.Client,
-	obj client.Object, finalizer string,
+	obj client.Object, finalizers ...string,
 ) error {
-	if !controllerutil.ContainsFinalizer(obj, finalizer) {
+	noFinalizersToPatch := true
```
Comment (garrettthomaskth) [inline internal/controllers/controllers.go:64]:
```suggestion
	if len(finalizers) == 0 {
		return nil
	}
```

File: config/images/package-loader.Containerfile (hunk 2 above)
Comment (yashvardhan-kukreja) [reply config/images/package-loader.Containerfile:1]: So, the package-loader image requires command like `mkdir` and `cp` to be executed against it which aren't supported under a scratch container. Therefore, I proceeded with the ubi-minimal container.

**Comment by yashvardhan-kukreja** [comment]
in favor of #67

**Pull request closed without merging**